/firefox/new/  title and description test

rraue opened the pull request:
## Description
Testing title and meta-description of /firefox/new/ with a switch, so the test can be turned off if results are negative
## Issue / Bugzilla link
https://github.com/mozilla/bedrock/issues/6188
## Testing

alexgibson commented on bedrock/firefox/templates/firefox/new/base.html:
```diff
@@ -10,18 +10,33 @@
+    {% endif %}
+    {%- endblock -%}
+    {% block page_title_suffix %}{% endblock %}
+    {% block page_desc %}{{_('Download Firefox for Windows, Mac, Linux, Android, and iOS. Firefox is an independent, people-first browser made by Mozilla. Get it now, it's free and fast. ✓')}}{% endblock %}
```
You need a curly apostrophe in `it’s` here. This is currently throwing an error

alexgibson commented on bedrock/firefox/templates/firefox/new/base.html:
```diff
@@ -10,18 +10,33 @@
+    {% block page_desc %}{{_('Download Firefox for Windows, Mac, Linux, Android, and iOS. Firefox is an independent, people-first browser made by Mozilla. Get it now, it's free and fast. ✓')}}{% endblock %}
+{% else %}
+    {# Bug 1438302 Avoid duplicate content for en-CA and en-GB pages. #}
+    {%- block page_title -%}
```
You can't include a block twice in a template like this. You will need to do the switch conditonal inside each block. I'll work on an example.

alexgibson commented on bedrock/firefox/templates/firefox/new/base.html:
```diff
@@ -10,18 +10,33 @@
+    {% elif LANG == 'en-GB' %}
+    The Fast and Free Web Browser 🦊 (UK)
+    {% else %}
+    {{_('The Fast and Free Web Browser 🦊')}}
```
This is breaking translations, you'll need to use the existing `Free Web Browser` string as a fallback.

alexgibson commented on bedrock/firefox/templates/firefox/new/base.html:
```diff
@@ -10,18 +10,33 @@
-{%- block page_title -%}
-  {% if LANG == 'en-CA' %}
+
+{% if switch('firefox_new_title_experiment') %}
```
It would be nice to include a comment here linking to the issue, as a clue for what this all means to future adventurers.

alexgibson commented on bedrock/firefox/templates/firefox/new/base.html:
```diff
@@ -10,18 +10,33 @@
+    {# Bug 1438302 Avoid duplicate content for en-CA and en-GB pages. #}
+    {%- block page_title -%}
+    {% if LANG == 'en-CA' %}
+    The Fast and Free Web Browser 🦊 (CA)
```
@pmac are there any encoding gotcha's about including emoji like this that you are aware of? I think this should be fine, but I may be unaware of edgecases.

pmac replied:
I'm actually not sure. It should be like any other unicode character in theory, so I'd say no, but I don't think we've done this exact thing before. So I'd say it "should" be okay, but we should keep an eye on it.

alexgibson replied:
Ok thanks. Yeah we're putting this behind a switch and only for a couple of locales. We can always flip it off if we notice anything weird. Thanks

alexgibson commented:
@rraue is this something you're still interesting in testing out?

rraue commented:
Yes, I still would like to test this, I am just not sure, if my coding skills are good enough for this. I will update the description, so it reflects the goal of this test and the changed title and description.

alexgibson commented:
@rraue happy to help finish this up here if you need

rraue commented:
@alexgibson That would be great, I will learn from your code. I don't know yet how to implement the switch right and the fallback you mentioned. I updated the description and changed there the wrong apostrophe as well.
